trieste/acquisition/function.py

```diff
@@ -2125,8 +2125,11 @@ def prepare_acquisition_function(
 
         :return: The determinant of the predictive function.
         """
-        if not hasattr(model, "predict_joint"):
-            raise AttributeError(
+        
+        try:
+            model.predict_joint(dataset.query_points[0:])
+        except NotImplementedError:
+            raise ValueError(
```

Comment:
Ah yeah we can try-catch inside the builder, was thinking doing this will be expensive compared to doing try-catch inside the function. my bad.

Reply:
I'm worried about that as well, it could be unnecessarily expensive - hm, probably we should move it to the acq function itself, it will raise it only on starting to acquire points but its worth the savings